kernel/src/callback.rs

```diff
@@ -6,41 +6,80 @@ pub struct AppId {
     idx: usize,
 }
 
+/// The kernel can masquerade as an app. IDs >= this value are the kernel.
+/// These IDs are used to identify which kernel container is being accessed.
+const KERNEL_APPID_BOUNDARY: usize = 100;
+
 impl AppId {
     pub fn new(idx: usize) -> AppId {
         AppId { idx: idx }
     }
 
+    pub const fn kernel_new(idx: usize) -> AppId {
+        AppId { idx: idx }
```

Comment:
> If you're not going to check, why have this function at all?

Current rust doesn't allow us to check statically. In the mean time, this at least makes intention clear.